Add components section (#17)
scaffold-eth/se2-docs: docs/components/inputs/Bytes32Input.md

---
sidebar_position: 2
---

# TODO-Bytes32Input

Display a `bytes32` input, having a conversion tool in the input to convert from `string` to `bytes32`.

![Bytes32Input Example](/img/Bytes32Input.png)

## Import

```ts
import { Bytes32Input } from "~~/components/scaffold-eth";
```

## Usage

```ts
const inputProps = {
  name: stateObjectKey,
  value: form?.[stateObjectKey],
  placeholder: paramType.name
    ? `${paramType.type} ${paramType.name}`
    : paramType.type,
  onChange: (value: any) => {
    setForm((form) => ({ ...form, [stateObjectKey]: value }));
  },
};

<Bytes32Input {...inputProps} />;
```

## Props

- `value` => You can enter a valid hexadecimal or you can input a `string` and use the conversion utility to be converted to `bytes32`.

> - `onChange` => Callback that is called when the bytes32 input data changes.

Reviewer: ```suggestion
- `onChange` => A callback invoked when the data in the bytes32 input changes
```

Author: Finally deleted this component from Docs
